### PR TITLE
GH-Action to check DB Entities

### LINKED ISSUE #4150: Create a script for verifying if database version was bumped if entities have been updated

### PR DESCRIPTION
**Pull request by JcMinarro**
### 🎯 Goal
Whenever an `@Entity` of our DB is updated, we need to update the structure of our DB. To do that on Room, we need to upgrade the Version number (And process any migration on the case it is needed).
If we don't upgrade it, a runtime exception is thrown when Room is instanciated

Fix #4150

### 🛠 Implementation details

_Describe the implementation_

### 🎨 UI Changes
A comment like this one is posted on every PR that contains any `@Entity` update
Screenshot from 2022-09-13 17-40-27

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: .github/workflows/check-entities.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          java-version: 11
+      - name: Find touched DB Entities
+        id: touchedEntities
+        continue-on-error: true
+        run: |
+          status=$(grep '@Entity' -rinl . | xargs -r git diff-index --name-only --exit-code --output=touchedEntities origin/${{ github.base_ref }} $(); echo $?)
+          output=$(cat touchedEntities)
+          output="${output//'%'/'%25'}"
+          output="${output//$'\n'/'%0A'}"
+          output="${output//$'\r'/'%0D'}"
```
Comment (JcMinarro) [inline .github/workflows/check-entities.yml:27]: Whenver we pass a multiline output from one step to another, it looses multiline structure, to fix it we need to replace some characters.
More info [here](https://github.com/orgs/community/discussions/26288)

File: .github/workflows/check-entities.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if: steps.touchedEntities.outcome == 'failure'
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: DB Entities has been updated, Do we need to upgrade DB Version
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v2
+        if: steps.touchedEntities.outcome == 'failure'
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
```
Comment (JcMinarro) [inline .github/workflows/check-entities.yml:43]: `steps.fc.outputs.comment-id` is the ID of any previous comment from the `github-actions[bot]` user on case it already commented on the PR.
If it already commented, the comment is updated

File: .github/workflows/check-entities.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+jobs:
+  check-db-entities:
+    name: Check DB Entities
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (JcMinarro) [inline .github/workflows/check-entities.yml:13]: Needed to fetch `develop` branch to be able to run `git diff` command

File: .github/workflows/check-entities.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          comment-author: 'github-actions[bot]'
+          body-includes: DB Entities has been updated, Do we need to upgrade DB Version
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v2
+        if: steps.touchedEntities.outcome == 'failure'
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            DB Entities has been updated, Do we need to upgrade DB Version?
```
Comment (adasiewiczr) [inline .github/workflows/check-entities.yml:46]:
```suggestion
            DB Entities have been updated. Do we need to upgrade DB Version?
```

File: .github/workflows/check-entities.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          echo ::set-output name=touchedEntities::$output
+          exit $status
+      - name: Find Comment
+        uses: peter-evans/find-comment@v2
+        id: fc
+        if: steps.touchedEntities.outcome == 'failure'
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: DB Entities has been updated, Do we need to upgrade DB Version
```
Comment (adasiewiczr) [inline .github/workflows/check-entities.yml:38]:
```suggestion
          body-includes: DB Entities have been updated. Do we need to upgrade DB Version?

```

**Review by adasiewiczr — APPROVED** [review]
Small suggestion, looks good besides that!

**Pull request merged**